src/sentry/relay/config/__init__.py

```diff
@@ -250,7 +256,7 @@ def _get_project_config(
 
     if features.has("organizations:metrics-extraction", project.organization):
         config["sessionMetrics"] = {
-            "version": 1,
+            "version": 2 if _should_extract_abnormal_mechanism(project) else 1,
```

Comment:
nit: Maybe we can create an enum or two constants for the extraction version, now that there are two possible values.